### DISCUSSION TITLE
ClickOnce Build

### DISCUSSION THREAD

**Discussion by TonyValenti** [cake-build/cake#3793] [Q&A]
How do I have Cake build/publish using ClickOnce?

**Comment by devlead** [comment]
I written a Cake recipe that might provide some inspiration on how to achieve that with Cake.

It's available on NuGet
https://www.nuget.org/packages/Cake.ClickOnce.Recipe/

Source here
https://github.com/devlead/Cake.ClickOnce.Recipe

And a blog post going through it a bit here
https://www.devlead.se/posts/2021/2021-03-03-introducing-cake-clickonce-recipe

**Reply by TonyValenti** [reply]
@devlead - I would love it if you could convert this to work with Cake!

**Reply by devlead** [reply]
> @devlead - I would love it if you could convert this to work with Cake!

It's a Cake recipe, so it's written as a Cake scripts packaged as a NuGet and already works with Cake using the [Load Directive](https://cakebuild.net/docs/writing-builds/preprocessor-directives/load#:~:text=The%20load%20directive%20is%20used,load%20cake%20scripts%20from%20nuget.) 

Example usage:

```cake 
#load "nuget:?package=Cake.ClickOnce.Recipe&version=0.6.0"

ClickOnce.ApplicationName = "MyApp";
ClickOnce.Publisher = "devlead";
ClickOnce.PublishUrl = "https://clickoncenet5test.blob.core.windows.net/publish";
ClickOnce.RunBuild();
```
